microcosm_flask/audit.py

```diff
@@ -2,80 +2,131 @@
 Audit log support for Flask routes.
 
 """
-from logging import getLogger
+from collections import namedtuple
 from functools import wraps
+from logging import getLogger
 from json import loads
 from traceback import format_exc
 
 from flask import current_app, g, request
-
+from microcosm.api import defaults
 from microcosm_flask.errors import (
     extract_context,
     extract_error_message,
     extract_status_code,
 )
 
 
+X_REQUEST = "X-Request"
+
+AuditOptions = namedtuple("AuditOptions", [
+    "include_request_body",
+    "include_response_body",
+    "include_header",
+])
+
+
+def include_header_matching_prefix(prefix):
+    """
+    Match all headers starting with prefix.
+
+    """
+    def include_header(header):
+        return header.startswith(prefix)
+    return include_header
+
+
 def audit(func):
```

Comment:
I've retained the original `audit` function because some projects that have not fully ported to microcosm import this directly.